packages/indexer/src/starknet/parser.ts

```diff
@@ -0,0 +1,56 @@
+import { FieldElement } from "./felt";
+import { EventFilter } from "./filter";
+import { EventWithTransaction } from "./block";
+
+import { type Abi, hash } from "starknet";
+
+interface EventFilterParser {
+  eventFilter(name: string): EventFilter;
+  eventParser(name: string): ({ event, transaction }: EventWithTransaction ) => any;
+}
+
+// The Contract class implements the EventFilterParser interface.
+export class Contract implements EventFilterParser {
+  // Read-only properties for the contract's address and its ABI.
+  readonly contractAddress: FieldElement;
+  readonly contractAbi: Abi;
+
+  // Constructor to initialize a Contract instance with a contract address and ABI.
+  constructor(contractAddress: FieldElement, contractAbi: Abi) {
+    this.contractAddress = contractAddress;
+    this.contractAbi = contractAbi;
+  }
+
+  // Method to filter events based on their name from the contract's ABI.
+  eventFilter(name: string): EventFilter {
+    // Find the event in the ABI matching the provided name.
+    const event: EventFilter = this.contractAbi.find((item) => item.name === name);
```

Comment:
Notice that [the `Abi` type](https://github.com/starknet-io/starknet.js/blob/develop/src/types/lib/contract/abi.ts#L2C31-L2C31) is defined as an array of `FunctionAbi | EventAbi | StructAbi`. We are interested in `EventAbi`, so this line becomes:

```suggestion
    // remember to import `EventAbi` from starknet
    const event: EventAbi | undefined = this.contractAbi
      .find((item) => item.type === "event" && item.name === name);
```

Reply:
Maybe this method can be refactored in a private method `findEvent` since it's needed by the event parser too.